File: flow/script/download_youtube.ts
#!/usr/bin/env -S deno run --allow-net --allow-read --allow-write --allow-run --allow-env
/**
 * Download a single YouTube video.
 *
 * Usage:
 *   deno run --allow-net --allow-read --allow-write --allow-run --allow-env \
 *     flow/script/download_youtube.ts <URL>
 */

import { PATHS } from "./config/paths.ts";
import { downloadVideo } from "./lib/download.ts";

const url = Deno.args[0];
if (!url) {
  console.error("Usage: download_youtube.ts <YOUTUBE_URL>");
  Deno.exit(1);
}

console.log(`[INFO] Target URL: ${url}`);
console.log(`[INFO] Download dir: ${PATHS.downloadDir}`);

try {
  await Deno.mkdir(PATHS.downloadDir, { recursive: true });
  const saved = await downloadVideo({ url, outputDir: PATHS.downloadDir });
  console.log(`[SUCCESS] Saved to: ${saved}`);
  Deno.exit(0);
} catch (e: any) {
  console.error(`[ERROR] ${e.message}`);
  Deno.exit(1);
}

File: flow/script/download_youtube.py
#!/usr/bin/env python3
"""
Download a single YouTube video.

Usage:
    python flow/script/download_youtube.py <URL>
"""

import argparse
import sys
from pathlib import Path

from config.paths import PATHS
from lib.download import download_video


def main() -> int:
    parser = argparse.ArgumentParser(description="Download YouTube video")
    parser.add_argument("url", help="YouTube video or playlist URL")
    args = parser.parse_args()

    download_dir = PATHS["download_dir"]
    download_dir.mkdir(parents=True, exist_ok=True)

    print(f"[INFO] Target URL: {args.url}")
    print(f"[INFO] Download dir: {download_dir}")

    try:
        saved = download_video(args.url, str(download_dir))
        print(f"[SUCCESS] Saved to: {saved}")
        return 0
    except Exception as e:
        print(f"[ERROR] {e}", file=sys.stderr)
        return 1


if __name__ == "__main__":
    sys.exit(main())
